fs-one-concepts.md

```diff
@@ -4,18 +4,62 @@ Try your best to answer each question on your own before looking up the answer o
 
 #### 1. What is Enzyme and what are some of the methods that it provides?
 
+Enzyme allows for you to do testimg while using react by making it easier to test your react components output data. It allows for you to manipulate and simulate runtime given the output.
+
+
 #### 2. What is the difference between dynamic and a static routes?
 
+The main difference between dynamic and static routes are that with dynamic routes you can make variable declaretions with (:) within a route. we can pass anything we like on that section of the URL, and it will be passed to the component as a prop. Static routes are more stagnant. 
+ 
 #### 3. What is a JSON API?
 
+JSON stands for JavaScript Object Notation. It is interchange that can happen between two computer applications at different locatations throughout the world while using the same hardware machine.
+
 #### 4. What is a migration and why would you use one?
 
+A react migration is used whenn you want to update the app without stopping the server from running or rewriting your entire front end. You can migrate the new components to react one small piece at a time.
+
 #### 5. Explain how to set up a route in React.
 
+To set up page routing in react you must first add a new npm package to our app called react-router-com:
+
+yarn add react-router-dom
+
+and in app.js:
+
+import React, { Component } fom "react";
+import reactDOM from 'react-dom'
+import {
+        BrowserRouter as Router,
+        Route,
+        Link 
+} from "react-router-dom";
+
+Routes have a path and a component prop. The path looks for a match in the browsers url and the component prop renders the page when there IS a match.
+
 #### 6. When would you use a generate resource over a generate controller?
 
+- using rails g controller with specific method names causes the generator to only map specific routes to the route file. 
+- rails g resources will assume that you want the whole resource functionalilty and will map RESOURCES.
+
+
 #### 7. Explain the difference between a controller spec and a request spec.
 
+request specs are designed to drive behavior through the full stack including routing
+
+controller specs allow for you to simulate a single http request in each example and then specificy expected outcomes. 
+
 #### 8. Describe the React component lifecycle. What are some of the lifecycle methods?
 
-#### 9. At this point in the program, what technologies/languages do you find yourself gravitating to?
\ No newline at end of file
+There are different lifecycle methods that react provides at different phases of a component's life. React automatically calls the responsible method according to the phase in which the component is. By doing so we have better control over our component and can mainpulate them through said methods. 
+
+Lifecycle methods include:
+- initialization
```

Comment:
Ah, close! You are actually listing some of the phases of the component lifecycle. Take a stab at listing some of the methods involved with the lifecycle (hint: the render() method is one of them. What are some of the others we've used?)